packages/daemon/src/daemon.js

```diff
@@ -1454,6 +1451,33 @@ const makeDaemonCore = async (
     throw new Error('Cannot connect to peer: no supported addresses');
   };
 
+  /**
+   * The "known peers store" is like a pet store, but maps node identifiers to
+   * full peer ids.
+   *
+   * @type {import('./types.js').DaemonCore['provideKnownPeers']}
+   */
+  const provideKnownPeers = async peersFormulaId => {
+    // "Known peers" is just a pet store with an adapter over it.
+    const petStore = /** @type {import('./types.js').PetStore} */ (
+      await provide(peersFormulaId)
+    );
+
+    // Pet stores do not accept full ids as names.
+    /** @param {string} nodeIdentifier */
+    const getNameFor = nodeIdentifier => {
+      return `p${nodeIdentifier.slice(0, 126)}`;
+    };
+
+    return harden({
+      has: nodeIdentifier => petStore.has(getNameFor(nodeIdentifier)),
+      identify: nodeIdentifier =>
+        petStore.identifyLocal(getNameFor(nodeIdentifier)),
+      write: (nodeIdentifier, peerId) =>
+        petStore.write(getNameFor(nodeIdentifier), peerId),
+    });
+  };
```

Comment:
Another angle for this would be to introduce a `known-peers-store` formula type that is just like `pet-store` except that it is constructed with a different `assertValidName`. The same trick will probably be necessary for message persistence.

Reply:
#2166